### PR TITLE
[MM-34144] Rewrite hello world to quickstart guide

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This PR rewrites the existing hello world guide into a quick start guide. It guides a new developers though the process of setting up the first app.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34144

### CODE REVIEW

**Review by justinegeffen — APPROVED** [review]
This is excellent, @hanzei. It covers a lot of what I'd added to the overview and explains it a lot better. Thank you!

File: site/content/integrate/apps/quick-start-go/_index.md [hunk 1]
```diff
@@ -5,4 +5,264 @@ description: This quick start guide will walk you through the basics of writing
...
+	"requested_locations": [
+		"/channel_header",
+		"/command"
+	]
+}
+```
+
+### Bindings and locations
+
+Locations are named elements in the Mattermost user interface. Bindings specify how  an app's calls should be displayed at, and invoked from, these locations.
```
Comment (aaronrothschild) [inline site/content/integrate/apps/quick-start-go/_index.md:74]:
```suggestion
Locations are named elements in the Mattermost user interface. Bindings specify how an app's calls should be displayed at, and invoked from, these locations.
```

File: site/content/integrate/apps/quick-start-go/_index.md [hunk 2]
```diff
@@ -5,4 +5,264 @@ description: This quick start guide will walk you through the basics of writing
...
+
+The app creates a channel header button, and adds a `/helloworld send` command.
```
Comment (aaronrothschild) [inline site/content/integrate/apps/quick-start-go/_index.md:76]:
```suggestion
The app manifest above, creates a channel header button, and adds a `/helloworld send` command.
```

**Review by aaronrothschild — APPROVED** [review]

File: site/content/integrate/apps/quick-start-go/_index.md [hunk 3]
```diff
@@ -5,4 +5,264 @@ description: This quick start guide will walk you through the basics of writing
...
+- Contains a `manifest.json`, declares itself an HTTP application that acts as a bot, and attaches to locations in the user interface.
+- Attaches `send-modal` in its `bindings` to a button in the channel header, and `send` to a `/helloworld` command.
+- Contains a `send` function that sends a parameterized message back to the user.
+- Contains a `send-modal` function that forces displaying the `send` form as a modal.
+
+## Prerequisites
+
+Before you can start with your app, you first need to set up a local developer environment following the [server](/contribute/server/developer-setup/) and [webapp](/contribute/webapp/developer-setup/) setup guides. You must enable the apps feature flag before starting the Mattermost server by setting the enjoinment variable `MM_FEATUREFLAGS_AppsEnabled` to `true` by e.g. adding `export MM_FEATUREFLAGS_AppsEnabled=true` to your `.bashrc`. Please also ensure that `Bot Account Creation` and `OAuth 2.0 Service Provider` are enabled in the System Console.
+
+**Note:** Apps do not work with a production release of Mattermost right now. They can only be run in a development environment.
```
[the remark this answers is not in the document]
Comment (hanzei) [reply site/content/integrate/apps/quick-start-go/_index.md:19]: The Apps plugin can be installed, but app bindings won't show. There is no easy way to prohibit the installation right now.

[the remark this answers is not in the document]
Comment (hanzei) [reply site/content/integrate/apps/quick-start-go/_index.md:19]: Added "A future release will support production environments"

File: site/content/integrate/apps/quick-start-go/_index.md [hunk 4]
```diff
@@ -5,4 +5,264 @@ description: This quick start guide will walk you through the basics of writing
...
+
+You also need at least `go1.16` installed. Please follow the guide [here](https://golang.org/doc/install) to install the latest version.
+
+### Install the Apps plugin
+
+The [apps plugin](https://github.com/mattermost/mattermost-plugin-apps)) is a communication bridge between your app and the Mattermost server. To install it on your local server by cloning the code in a directory of your choice:
```
[the remark this answers is not in the document]
Comment (hanzei) [reply site/content/integrate/apps/quick-start-go/_index.md:25]: Not really right now. We are iterating quickly. Once we have a stable release, I'm happy to link it here.

**Pull request merged**